config/routes.rb

```diff
@@ -2,60 +2,81 @@
+  # '/app_settings/...'
+  resources :app_settings, only: %i[update edit]
+  # '/hosts/...', '/servers/...'
+  resources :hosts, :servers
+  get '/hosts/:id' => 'hosts#show', constraints: { id: /.*/ }
 
+  # '/notifications/...'
   resources :notifications, only: %i[index new create destroy] do
-    get :mark_as_read, on: :member
+    member do
+      get :mark_as_read
+      delete :destroy_and_redirect
+    end
     get :has_any, on: :collection, to: 'notifications#any?'
-    delete :destroy_and_redirect, on: :member
   end
 
-  get '/dashboard' => 'dashboard#index', as: :dashboard
-  root to: 'landing#index'
-
-  get '/hosts/:id' => 'hosts#show', constraints: { id: /.*/ }
-
-  get '/vms/configs/:id' => 'vms#edit_config', constraints: { id: /.*/ }, as: :edit_config
-  patch '/vms/configs/:id' => 'vms#update_config', constraints: { id: /.*/ }, as: :update_config
-
-  # move all vm actions under /vms/vm because a VM named requests might otherwise lead to issues!
-  post '/vms/vm/:id/change_power_state' => 'vms#change_power_state', constraints: { id: /.*/ }
-  post '/vms/vm/:id/suspend_vm' => 'vms#suspend_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/shutdown_guest_os' => 'vms#shutdown_guest_os', constraints: { id: /.*/ }
-  post '/vms/vm/:id/reboot_guest_os' => 'vms#reboot_guest_os', constraints: { id: /.*/ }
-  post '/vms/vm/:id/reset_vm' => 'vms#reset_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/request_vm_archivation' => 'vms#request_vm_archivation', constraints: { id: /.*/ }
-  post '/vms/vm/:id/archive_vm' => 'vms#archive_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/request_vm_revive' => 'vms#request_vm_revive', constraints: { id: /.*/ }
-  post '/vms/vm/:id/revive_vm' => 'vms#revive_vm', constraints: { id: /.*/ }
-  post '/vms/vm/:id/stop_archiving' => 'vms#stop_archiving', constraints: { id: /.*/ }
-  resources :vms, except: %i[new create], path: 'vms/vm'
-
-  get 'slack/new' => 'slack#new', as: :new_slack
-  get 'slack/auth' => 'slack#update', as: :update_slack
-
-  devise_for :users,
-             path: 'users',
-             controllers: {
-               registrations: 'users/registrations',
-               omniauth_callbacks: 'users/omniauth_callbacks'
-             }
+  # '/slack/...'
+  scope 'slack' do
+    get 'new' => 'slack#new', as: :new_slack
+    get 'auth' => 'slack#update', as: :update_slack
+  end
 
-  resources :hosts, :servers
-  resources :users do
-    member do
-      patch :update_role
+  # https://guides.rubyonrails.org/routing.html#prefixing-the-named-route-helpers
+  # '/vms/...'
+  scope 'vms' do
+    # '/vms/request_templates/...'
+    resources :request_templates, except: :show
+    # '/vms/requests/operating_systems/...'
+    # Order matters here, as routes are matched from top to bottom
+    # If 'resources :requests' is first, it will attempt to match that
+    resources :operating_systems, path: 'requests/operating_systems', except: :show
```

Comment:
Why is this nested under vms?

Reply:
That was the configuration before the refactoring. We can totally change this, maybe not in this refactoring PR though. I'll have to check if any tests / code relies on the OSs being available under that path.

Reply:
Related ticket: https://github.com/hpi-swt2/vm-portal/issues/480